### ISSUE TITLE
Route agent crash when handling removed local endpoint

### ISSUE THREAD

**Issue by skitt** [submariner-io/submariner#2573]
This is extracted from the logs in #2572 but probably not related to the connection downtime.

The route agent crashed with

```
E0623 12:25:05.095498       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 823 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x14f24a0, 0x21db0b0)
	k8s.io/apimachinery@v0.21.0/pkg/util/runtime/runtime.go:74 +0xa6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.21.0/pkg/util/runtime/runtime.go:48 +0x86
panic(0x14f24a0, 0x21db0b0)
	runtime/panic.go:965 +0x1b9
github.com/submariner-io/submariner/pkg/routeagent_driver/handlers/ovn.(*Handler).handleSubnets(0xc000194840, 0xc000f39d60, 0x2, 0x2, 0x173d9e0, 0x173f468, 0x55, 0xb7)
	github.com/submariner-io/submariner/pkg/routeagent_driver/handlers/ovn/south_rules.go:35 +0x5f
github.com/submariner-io/submariner/pkg/routeagent_driver/handlers/ovn.(*Handler).cleanupGatewayDataplane(0xc000194840, 0xc000e254a0, 0x1)
	github.com/submariner-io/submariner/pkg/routeagent_driver/handlers/ovn/gateway_dataplane.go:40 +0x99
github.com/submariner-io/submariner/pkg/routeagent_driver/handlers/ovn.(*Handler).TransitionToNonGateway(0xc000194840, 0x0, 0x0)
	github.com/submariner-io/submariner/pkg/routeagent_driver/handlers/ovn/handler.go:201 +0x70
github.com/submariner-io/submariner/pkg/event.(*Registry).TransitionToNonGateway.func1(0x188f5f8, 0xc000194840, 0x0, 0x0)
	github.com/submariner-io/submariner/pkg/event/registry.go:94 +0x34
github.com/submariner-io/submariner/pkg/event.(*Registry).invokeHandlers(0xc0003d4480, 0x16a1923, 0x16, 0x173d9c8, 0x11e6aa0, 0xc0003176c0)
	github.com/submariner-io/submariner/pkg/event/registry.go:162 +0xba
github.com/submariner-io/submariner/pkg/event.(*Registry).TransitionToNonGateway(0xc0003d4480, 0xc000716630, 0x22)
	github.com/submariner-io/submariner/pkg/event/registry.go:93 +0x4c
github.com/submariner-io/submariner/pkg/event/controller.(*Controller).handleRemovedLocalEndpoint(0xc000448180, 0xc0003176c0, 0x0, 0x0)
	github.com/submariner-io/submariner/pkg/event/controller/endpoint_removed.go:53 +0x105
```

The relevant code is https://github.com/submariner-io/submariner/blob/v0.12.0/pkg/routeagent_driver/handlers/ovn/south_rules.go#L35 but it would seem that `ovn.localEndpoint` is `nil` (see also https://github.com/submariner-io/submariner/blob/v0.12.0/pkg/routeagent_driver/handlers/ovn/handler.go#L123).

**Comment by dfarrell07** [comment]
This still exists in 0.13.6, but is fixed in 0.14 and beyond.

https://github.com/submariner-io/submariner/blob/v0.13.6/pkg/routeagent_driver/handlers/ovn/south_rules.go#L36

**Comment by tpantelis** [comment]
This was recently fixed by removing the `ovn.localEndpoint` field.
